File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(br.readLine());

        StringTokenizer st = new StringTokenizer(br.readLine());

        Set<Integer> aSet = new HashSet<>();

        for (int i = 0; i < n; i++) {
            aSet.add(Integer.parseInt(st.nextToken()));
        }

        int m = Integer.parseInt(br.readLine());

        int[][] mArr = new int[m][2];

        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < m; i++) {
            mArr[i][0] = Integer.parseInt(st.nextToken());
        }

        for (int i = 0; i < m; i++) {
            if (aSet.contains(mArr[i][0])) {
                mArr[i][1] = 1;
            }
            bw.write(mArr[i][1] + "\n");
        }

        bw.flush();
        br.close();
        bw.close();


    }
}

File: 백준/Silver/1920. 수 찾기/수 찾기.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.lang.StringBuilder
import java.util.StringTokenizer

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val sb = StringBuilder()
    val n = Integer.parseInt(br.readLine())
    val nList = mutableSetOf<Int>()
    var st = StringTokenizer(br.readLine())
    for (i in 0 until n){
        nList.add(Integer.parseInt(st.nextToken()))
    }
    val m = Integer.parseInt(br.readLine())
    val mList = Array(m,{ Array(2,{0})})
    st = StringTokenizer(br.readLine())
    for (i in 0 until m){
        mList[i][0] = Integer.parseInt(st.nextToken())
    }

    for (i in 0 until m){
        if (nList.contains(mList[i][0])) mList[i][1] = 1
        sb.append("${mList[i][1]}").append("\n")
    }
    println(sb)
}